service/src/java/org/apache/hive/service/auth/HiveAuthFactory.java

```diff
@@ -145,6 +145,11 @@ public HiveAuthFactory(HiveConf conf) throws TTransportException {
 
         delegationTokenManager.startDelegationTokenSecretManager(conf, baseHandler, ServerMode.HIVESERVER2);
         saslServer.setSecretManager(delegationTokenManager.getSecretManager());
+
```

Comment:
Could this change L148-L152 be incorporated into the saslServer itself? Or the classname could be passed into the HiveAuthFactory from within the createCustomWithKerberos runnable at L218?